### ISSUE TITLE
beforeException dispatcher forward not works on exception from plugin

### ISSUE THREAD

**Issue by tugrul** [phalcon/cphalcon#12542]
### Expected and Actual Behavior

> dispatcher not dispatching forwarded action from beforeException event.

https://github.com/tugrul/invo/commit/2ad34a0ef102b9509ae86000cdcc6c7a4374325d

beforeException able to handle that exception but $dispatcher->forward() not works

https://github.com/tugrul/invo/blob/master/app/plugins/NotFoundPlugin.php#L41

### Details
* Phalcon version: 3.0.2
* PHP Version: PHP 7.0.14-1~dotdeb+8.1
* Operating System: Debian 8
* Installation type: Compiling from source
* Server: Nginx + PHP-FPM

**Comment by guweigang** [comment]
HI, I also met this problem today in `Phalcon 3.2.0`.  You can `dispatch()` it  yourself after `forward()`. Or you can change codes in https://github.com/phalcon/cphalcon/blob/master/phalcon/dispatcher.zep#L656  to 
```php
if this->{"_handleException"}(e) === false {
	continue;
} elseif this->_finished === false {
        continue;
} else {
	throw e;
}
```

Maybe that is the true purpose in `catch` @sergeyklay .

**Comment by guweigang** [comment]
any progress？

_Sent from my OnePlus ONEPLUS A5000 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by virgofx** [comment]
Related to #12931. We have a PR that needs to rebase that fixes all the initialize/beforeException events.

**Comment by guweigang** [comment]
@virgofx yes，We could not catch exceptions in actions here before, and now we can, but we can not process `dispatcher->forward` correctly

_Sent from my OnePlus ONEPLUS A5000 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by sergeyklay** [comment]
Fixed in the `3.2.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
